climada/entity/exposures/crop_production.py

```diff
@@ -218,7 +250,20 @@ def set_from_single_run(self, input_dir=INPUT_DIR, filename=None, hist_mean=HIST
 
         # set historic mean, its latitude, and longitude:
         hist_mean_dict = dict()
-        if isdir(hist_mean):
+        # if hist_mean is given as np.ndarray or dict,
+        # code assumes it contains hist_mean as returned by the hazard crop_potential
+        # however structured in dictionary as hist_mean_dict, with same
+        # bbox extensions as the exposure:
+        if isinstance(hist_mean, dict):
+            if not ('firr' in hist_mean.keys() or 'noirr' in hist_mean.keys()):
```

Comment:
What about `combined`?

Reply:
hist_mean is never `combined`, it always provides either 'firr' or 'noirr' data. If users choose `irr='combined'`, the hist_mean is loaded for each irrigation type and combined further below:

In lines 238f, the input irr='combined' is translated into the two combination types:
```
        if irr == 'combined':
            irr = ['firr', 'noirr']
```

in lines 304f, they are combined:
```
        for irr_val in irr[1:]: # add other irrigation types if irr=combined
            value_tmp = np.squeeze(area_crop[irr_val]*hist_mean_dict[irr_val][idx_mean])
            value_tmp = np.nan_to_num(value_tmp) # replace NaN by 0.0
            self['value'] += value_tmp
```

should this be explained better with comments?

Reply:
You know that I am always in favor of good comment :D